### PR TITLE
controller: Return a more descriptive error for duplicate deployment.

### PR DESCRIPTION
**Pull request by archSeer**
Signed-off-by: Blaž Hrastnik <blaz.hrast@gmail.com>

### CODE REVIEW

File: controller/deployment.go [hunk 1]
```diff
@@ -138,6 +138,13 @@ func (c *controllerAPI) CreateDeployment(ctx context.Context, w http.ResponseWri
 		Strategy:     app.Strategy,
 	}
 	if err := c.deploymentRepo.Add(deployment); err != nil {
+		if e, ok := err.(*pq.Error); ok && e.Code.Name() == "unique_violation" && e.Constraint == "isolate_deploys" {
+			httphelper.Error(w, httphelper.JSONError{
+				Code:    httphelper.UnknownError,
+				Message: "Deploy exists for app (only one may be active at once)",
```
Comment (titanous) [inline controller/deployment.go]: > Cannot create deploy, there is already one in progress for this app.

And it should be a `validation_error`.

**Comment by archSeer** [comment]
@titanous Updated.

**Comment by titanous** [comment]
Please add a test for this.

**Comment by titanous** [comment]
LGTM.

**Pull request merged**